sdk/monitor/monitor-opentelemetry-exporter/src/utils/eventhub.ts

```diff
@@ -46,20 +46,22 @@ export const parseEventHubSpan = (span: ReadableSpan, envelope: Envelope): void
     "unknown") as string).replace(/\/$/g, ""); // remove trailing "/"
   const messageBusDestination = (span.attributes[MessageBusDestination] || "unknown") as string;
 
+  const baseData = envelope.data.baseData as RemoteDependencyData;
+
   switch (span.kind) {
     case SpanKind.CLIENT:
-      envelope.data.baseData.type = namespace;
-      envelope.data.baseData.target = `${peerAddress}/${messageBusDestination}`;
+      baseData.type = namespace;
+      baseData.target = `${peerAddress}/${messageBusDestination}`;
       break;
     case SpanKind.PRODUCER:
-      envelope.data.baseData.type = `Queue Message | ${namespace}`;
-      envelope.data.baseData.target = `${peerAddress}/${messageBusDestination}`;
+      baseData.type = `Queue Message | ${namespace}`;
+      baseData.target = `${peerAddress}/${messageBusDestination}`;
       break;
     case SpanKind.CONSUMER:
-      envelope.data.baseData.type = `Queue Message | ${namespace}`;
-      envelope.data.baseData.source = `${peerAddress}/${messageBusDestination}`;
-      envelope.data.baseData.measurements = {
-        ...envelope.data.baseData.measurements,
+      baseData.type = `Queue Message | ${namespace}`;
+      (baseData as any).source = `${peerAddress}/${messageBusDestination}`;
```

Comment:
`baseData` is currently generated as an empty interface for each Envelope type. I need to investigate whether the TS extension or the swagger file should be improved to make these nicer to work with

Reply:
I'd hope we can generate something better than empty! @srnagar - any thoughts on how the swagger can be improved?